packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/method_channel/method_channel_firebase_dynamic_links.dart

```diff
@@ -0,0 +1,192 @@
+// Copyright 2021, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:firebase_dynamic_links_platform_interface/firebase_dynamic_links_platform_interface.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/services.dart';
+
+import 'utils/exception.dart';
+
+/// The entry point for accessing a Dynamic Links instance.
+///
+/// You can get an instance by calling [FirebaseDynamicLinks.instance].
+class MethodChannelFirebaseDynamicLinks extends FirebaseDynamicLinksPlatform {
+  /// Create an instance of [MethodChannelFirebaseDynamicLinks] with optional [FirebaseApp]
+  MethodChannelFirebaseDynamicLinks({FirebaseApp? app})
+      : super(appInstance: app) {
+    channel.setMethodCallHandler((MethodCall call) async {
```

Comment:
Should this be setting a new handler each time class is instantiated? Might be worth adding a static value to check if it's already been set. Calling this will replace any previous handlers, and there might be a very small chance an event comes through whilst the handler is being set. Unlikely but possible.